testTruckRender.gd

```diff
@@ -0,0 +1,56 @@
+@tool
+extends Sprite2D
+
+var truckModel;
+var postProcObject;
+var lastPos: Vector2;
+var currentPos: Vector2;
+var extend_vec: Vector2;
+
+var angle_to = 0;
+
+var lastTick = 0
+
+@export var modelTint: Color = Color.RED:
+	set(val):
+		modelTint = val;
+@export_range(0, 1, 0.1) var modelTintFactor:
+	set(val):
+		modelTintFactor = val;
+		
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	truckModel = $SubViewport/testTruck
+	postProcObject = get_node("SubViewport/Camera3D/Postprocess")
+	pass # Replace with function body.
+
+func _draw():
+#	if !Engine.is_editor_hint():
+		draw_line(position, position + (extend_vec), Color.RED, 2);
+		draw_arc(position, 20, 0, deg_to_rad(-1 * angle_to), 20, Color.YELLOW, 2)
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	if postProcObject:
+		postProcObject.get_surface_override_material(0).set_shader_parameter("tint_color", modelTint)
+		postProcObject.get_surface_override_material(0).set_shader_parameter("tint_effect_factor", modelTintFactor)
+	if !Engine.is_editor_hint():
+		var controlNode = get_tree().get_current_scene().get_node('ControlSet')
+		if controlNode.get_node('VBoxContainer/PauseUnpause') and controlNode.get_node("VBoxContainer/PauseUnpause").isPaused:
+			return
+		if controlNode.get_node("VBoxContainer/HScrollBar") and controlNode.get_node("VBoxContainer/HScrollBar").value <= 0.01:
+			return
+	
+	if get_tree().get_current_scene():
+		if !currentPos:
+			currentPos = get_parent().position
+	
+		lastPos = currentPos;
+		currentPos = get_parent().global_position
+		extend_vec = lastPos.direction_to(currentPos) * 20
```

Comment:
Since you just set `lastPos` to `currentPos`, how does this work?

Reply:
lastPos is set to the old currentPos from the last processing tick, currentPos is updated to the parent's global_position on line 49

Reply:
Oh, I just didn't read this correctly. My bad.